go-controller/pkg/clustermanager/status_manager/status_manager.go

```diff
@@ -0,0 +1,122 @@
+package status_manager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	adminpolicybasedrouteapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/adminpolicybasedroute/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+// fieldManagerName should be different from any zone name
+const fieldManagerName = "cluster-manager"
+
+type typedStatusManager[T Object] struct {
+	informer cache.SharedIndexInformer
+	// cluster-scoped resources should ignore namespace
+	patch func(ctx context.Context, namespace, name string, pt ktypes.PatchType, data []byte, opts metav1.PatchOptions,
+		subresources ...string) (T, error)
+	// updateStatusFunc should handle nil in case of pointer type, and return (updatedObject, needsUpdate).
+	// When possible, return a new object with just the right status set, this makes encoding faster.
+	updateStatusFunc func(obj T) ([]byte, bool)
+}
+
+type Object interface {
+	runtime.Object
+	metav1.Object
+}
+
+func (m *typedStatusManager[T]) updateStatus(oldObjInterface, newObjInterface interface{}) {
+	oldObj, ok := oldObjInterface.(T)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("expecting %T but received %T", *new(T), oldObj))
+		return
+	}
+	newObj, ok := newObjInterface.(T)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("expecting %T but received %T", *new(T), newObj))
+		return
+	}
+
+	patch, needsUpdate := m.updateStatusFunc(newObj)
+	if !needsUpdate {
+		return
+	}
+
+	patchOptions := metav1.PatchOptions{
+		Force:        utilpointer.Bool(true),
+		FieldManager: fieldManagerName,
+	}
+	_, err := m.patch(context.TODO(), newObj.GetNamespace(), newObj.GetName(), ktypes.ApplyPatchType, patch, patchOptions, "status")
+
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("could not patch: %w", err))
```

Comment:
do you want to list the object namespaced name that cannot be patched? or are you expected that to be included in the err?